Crafttweaker_Annotation_Processors/src/main/java/com/blamejared/crafttweaker_annotation_processors/processors/validation/PreprocessorAnnotationValidationProcessor.java

```diff
@@ -32,6 +31,27 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
                     this.processingEnv.getMessager()
                             .printMessage(Diagnostic.Kind.ERROR, "Element is annotated as Preprocessor but is not assignable to " + preprocessorInterfaceCanonicalName + "!", element);
                 }
+    
+                //Not sure if the implicit no-arg constructor is already present here,
+                // so let's just check no other constructor was found
+                boolean anyConstructorFound = false;
+                boolean noArgPublicConstructorFound = false;
+                for(Element enclosedElement : element.getEnclosedElements()) {
+                    if(enclosedElement.getKind() == ElementKind.CONSTRUCTOR) {
+                        anyConstructorFound = true;
+                        ExecutableElement constructor = (ExecutableElement) enclosedElement;
+                        final boolean noArg = constructor.getParameters().size() == 0;
+                        final boolean isPublic = constructor.getModifiers().contains(Modifier.PUBLIC);
+                         if(noArg && isPublic) {
+                             noArgPublicConstructorFound = true;
+                             break;
+                         }
+                    }
+                }
+                
+                if(!noArgPublicConstructorFound && anyConstructorFound){
+                    this.processingEnv.getMessager().printMessage(Diagnostic.Kind.ERROR, "Element is annotated as Preprocessor but has not public no-arg constructor!", element);
```

Comment:
"but has not public"

should be
"but has no public"

Reply:
Fixed ^^
Though the error will only come up at compiletime anyways